service/matching/taskQueueManager.go

```diff
@@ -510,8 +516,6 @@ func (c *taskQueueManagerImpl) executeWithRetry(
 
 	if _, ok := err.(*persistence.ConditionFailedError); ok {
```

Comment:
Instead of check error and unload all over the place, can we do it here just in one place (with that proposed func that checks error and unload)?

Reply:
This metric was probably wrong (too low). There are a lot of other places that use this error type where we weren't updating the counter. I moved this metric update inline with the `signalIfFatal` which should see _all_ `ConditionFailedError`s on their way through.